### PR TITLE
feat: Runtime Params

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Common/CIPipeline.Types.ts [hunk 1]
```diff
@@ -97,11 +97,34 @@ export enum RefVariableStageType {
     POST_CI = 'POST_CI',
 }
 
+export interface FileProperty {
```
Comment (eshankvaish) [inline src/Common/CIPipeline.Types.ts:100]: Please add Type to the interfaces across PR

File: src/Common/CIPipeline.Types.ts [hunk 2]
```diff
@@ -110,6 +133,12 @@ export interface VariableType {
     refVariableName: string
     refVariableStage?: RefVariableStageType
     variableStepIndexInPlugin?: number
+    fileMountDir: string
+    fileReferenceId: number
+    valueConstraintId: number
+    valueConstraint: ValueConstraint
+    isRuntimeArg: boolean
+    refVariableUsed: boolean
```
Comment (eshankvaish) [inline src/Common/CIPipeline.Types.ts:141]: Suggestion: Add ts docs to the types

File: src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    maxUploadSize,
+    abortControllerRef,
+}: {
+    file: File[]
+    appId: number
+    envId: number
+    allowedExtensions?: string[]
+    maxUploadSize?: number
+    abortControllerRef?: MutableRefObject<AbortController>
+}): Promise<UploadFileDTO> => {
```
Comment (eshankvaish) [inline src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts:22]: - allowedExtensions, maxUploadSize are already in one of the types. pls reuse
- abortControllerRef should be read from APIOptions

File: src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            { abortControllerRef },
+            true,
+        )
+
+        return result
+    } catch (err) {
+        if (getIsRequestAborted(err)) {
+            return null
+        }
+        showError(err)
```
Comment (eshankvaish) [inline src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts:42]: Not sure but i think we should throw the error even if it is aborted. Let me know if I'm missing something

File: src/Shared/Components/FileUpload/FileUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                    <div className="flexbox dc__align-items-center dc__gap-8 px-8 py-4 min-w-0">
+                        <ICCloudUpload className="icon-dim-16 dc__no-shrink" />
+                        <Tooltip content={fileName}>
+                            <span className="fs-12 lh-20 dc__ellipsis-right">{fileName}</span>
+                        </Tooltip>
+                    </div>
+                    <button
+                        type="button"
+                        className="dc__transparent flex p-6 dc__hover-n50"
+                        onClick={() => onUpload([])}
```
Comment (eshankvaish) [inline src/Shared/Components/FileUpload/FileUpload.tsx:41]: move the inline function

File: src/Shared/Components/FileUpload/FileUpload.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            const filesArray = Array.from(event.target.files)
+            onUpload(filesArray)
+        }
+        // Clearing value otherwise we can't upload the same file again.
+        const e = event
+        e.target.value = ''
+    }
+
+    return (
+        <div className={className || ''}>
```
Comment (eshankvaish) [inline src/Shared/Components/FileUpload/FileUpload.tsx:29]: We can remove the classname prop and wrap with a div

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts (hunk 3 above)
Comment (eshankvaish) [reply src/Pages/Applications/DevtronApps/Details/CDPipeline/services.ts]: abortControllerRef part seems to be left

File: src/Shared/Components/DynamicDataTable/styles.scss [hunk 7]
```diff
@@ -65,8 +65,9 @@
     &__cell {
         min-width: 0;
 
-        .select-picker-borderless-control {
-            padding: 8px;
+        &__select-picker__control {
```
Comment (eshankvaish) [inline src/Shared/Components/DynamicDataTable/styles.scss:68]: is this class targeting correctly?

**Review by eshankvaish — APPROVED** [review]

File: src/Shared/Components/DynamicDataTable/styles.scss (hunk 7 above)
Comment (RohitRaj011) [reply src/Shared/Components/DynamicDataTable/styles.scss:68]: yes

File: src/Common/TippyCustomized.tsx [hunk 8]
```diff
@@ -43,17 +43,19 @@ export const TippyCustomized = (props: TippyCustomizedProps) => {
     }
 
     const closeTippy = (e) => {
-        stopPropagation(e)
-        if (tippyRef.current?.hide) {
-            tippyRef.current.hide()
-            tippyRef.current = null
+        if (!props.disableClose) {
+            stopPropagation(e)
```
Comment (vivek-devtron) [inline src/Common/TippyCustomized.tsx:47]: Should this be outside of `if`?

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**